llvm/lib/Transforms/Utils/CallPromotionUtils.cpp

```diff
@@ -572,6 +575,89 @@ CallBase &llvm::promoteCallWithIfThenElse(CallBase &CB, Function *Callee,
   return promoteCall(NewInst, Callee);
 }
 
+CallBase *llvm::promoteCallWithIfThenElse(CallBase &CB, Function &Callee,
+                                          PGOContextualProfile &CtxProf) {
+  assert(CB.isIndirectCall());
+  if (!CtxProf.isFunctionKnown(Callee))
```

Comment:
Might be useful to have some statistics on how many promoted / dropped?

Reply:
That seems to belong in the pass that will exercise this, rather.